nilearn/plotting/displays/_slicers.py

```diff
@@ -395,6 +395,16 @@ def _map_show(
             img = reorder_img(img, resample=resampling_interpolation)
         threshold = float(threshold) if threshold is not None else None
 
+        if threshold is not None:
+            data = safe_get_data(img, ensure_finite=True)
+            if threshold == 0:
+                data = np.ma.masked_equal(data, 0, copy=False)
+            else:
+                data = np.ma.masked_inside(
+                    data, -threshold, threshold, copy=False
+                )
+            img = new_img_like(img, data, img.affine)
```

Comment:
@michellewang do you remember why this got removed in #3993 ?

I can see you refactored part of it into a new method used later in the code but I cannot quite see if readding this code now will interfere with something.

Reply:
@Remi-Gau I tried plotting some things and I don't see anything obviously wrong. Since the lines you added back only address the threshold part of the masking (not the `vmin`/`vmax` parts of the `_threshold` function I added), I don't think it would break anything. However, it might be confusing/not optimal to have 2 places in a function where thresholding/masking is done? I checked and the reason why I moved the thresholding further down in that function is because, for glass brain plotting with custom `vmin`, the thresholding needed to be done after we take the absolute value of the data if `plot_abs` is `True` (otherwise we incorrectly lose whatever is lower than `vmin`), but for some reason the absolute value is only taken in [`GlassBrainAxes.transform_to_2d`](https://github.com/nilearn/nilearn/blob/0bf5f707d9f2fa10021fef1994acffc03a44e08e/nilearn/plotting/displays/_axes.py#L438), which gives the 2D views to be plotted in each panel. It doesn't look like there is currently an easy way to take the absolute value of the whole 3D data (possibly because `plot_abs` is only for glass brain plots?), but maybe we can add a `plot_abs` parameter to the slicer objects and do the absolute value handling inside `_map_show`?

Anyway if this needs to be fixed as soon as possible I think your fix should be fine, but in the future we might want to consider not having the double thresholding/masking part.

Reply:
I think you're right. Passing the argument below sounds like the best fix.

Reply:
I'd say that we merge this as is and do the refactoring suggested by @michellewang in another PR to keep things easier to review